expertsystem/data.py

```diff
@@ -32,12 +33,29 @@ def __eq__(self, other: object) -> bool:
             return self.__value == other.value
         return self.__value == other
 
+    def __gt__(self, other: Any) -> bool:
+        if isinstance(other, Parity):
+            return self.value > other.value
+        return self.value > int(other)
+
+    def __ge__(self, other: Any) -> bool:
+        return self > other or self == other
+
+    def __lt__(self, other: Any) -> bool:
+        return not self >= other
```

Comment:
im not sure about standard python procedure here, but I'm quite used to just swapping the operands, so here
`self < other` -> `other > self`

Reply:
You mean `self < other` instead of `not self >= other`? But that would cause an infinite recursion

Reply:
No so the general idea is you define either `<` or `>`, and then you just switch the arguments if you want the opposite.
So if you defined greater than (>) then you would define less than (A < B) as `return B > A`.

Reply:
Not sure what you mean. The methods implemented here were required to get the following tests working:
https://github.com/ComPWA/expertsystem/blob/148b9419d5001f5f18447ea2346699052c2d33e7/tests/unit/test_data.py#L81-L89

Reply:
I'm saying
```python3
def __lt__(self, other: Any) -> bool:
        return other > self
```
looks better than
```python3
def __lt__(self, other: Any) -> bool:
        return not self >= other
```

Reply:
Ah ok. But still, it causes a max recursion:
```
expertsystem/data.py:45: in __lt__
    return other > self
expertsystem/data.py:45: in __lt__
    return other > self
E   RecursionError: maximum recursion depth exceeded in comparison
!!! Recursion detected (same locals & position)
```
(this is done on 148b9419 with the exact replacement as you describe)

Reply:
Does not make sense to me. I would say you made a mistake in your implementation then. So you should define 
```python3
def __gt__(self, other: Any) -> bool:
        return self.value > other.value 
```
(Note i simplified the implementation)
So not keep pointing one function to another. So you define one function and make all the other ones point to this one

Reply:
ca40fb25

Reply:
Ok just for clarification: the reason for the endless loop was the comparison for an `int` with a `Parity`